Задание 1.
Даны значения величины заработной платы заемщиков банка (zp) и значения их поведенческого кредитного скоринга (ks):
zp = [35, 45, 190, 200, 40, 70, 54, 150, 120, 110],
ks = [401, 574, 874, 919, 459, 739, 653, 902, 746, 832].
Найдите ковариацию этих двух величин с помощью элементарных действий, а затем с помощью функции cov из numpy
Полученные значения должны быть равны.
Найдите коэффициент корреляции Пирсона с помощью ковариации и среднеквадратичных отклонений двух признаков,
а затем с использованием функций из библиотек numpy и pandas.

In [106]:
import numpy as np
import pandas as pd

In [107]:
zp = np.array([35, 45, 190, 200, 40, 70, 54, 150, 120, 110])
ks = np.array([401, 574, 874, 919, 459, 739, 653, 902, 746, 832])


Ковариация через встроенную функцию.

In [134]:
np.cov(zp, ks, ddof=0)

array([[ 3494.64,  9157.84],
       [ 9157.84, 30468.89]])

Ковариация вручную.

In [135]:
cov = round((np.mean(zp*ks) - np.mean(zp) * np.mean(ks)), 2)
cov

9157.84

Коэффициент корреляции Пирсона вручную.

In [136]:
sigma_zp = round(np.std(zp, ddof=0), 2)
sigma_zp

59.12

In [137]:
sigma_ks = round(np.std(ks, ddof=0), 2)
sigma_ks

174.55

In [138]:
kor = cov / (sigma_zp * sigma_ks)
kor

0.8874395361899088

Коэффициент корреляции Пирсона через встроенные функции.

In [139]:
np.corrcoef(zp, ks)

array([[1.        , 0.88749009],
       [0.88749009, 1.        ]])

In [140]:
d = [zp, ks]
d

[array([ 35,  45, 190, 200,  40,  70,  54, 150, 120, 110]),
 array([401, 574, 874, 919, 459, 739, 653, 902, 746, 832])]

In [141]:
df = pd.DataFrame(data=d)
df

0    1    2    3    4    5    6    7    8    9
0   35   45  190  200   40   70   54  150  120  110
1  401  574  874  919  459  739  653  902  746  832

In [142]:
df.corr()

0    1    2    3    4    5    6    7    8    9
0  1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0
1  1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0
2  1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0
3  1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0
4  1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0
5  1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0
6  1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0
7  1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0
8  1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0
9  1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0  1.0

Через Pandas я так и не понял как правильно метод .corr() использовать

Задание 2.
Измерены значения IQ выборки студентов,
обучающихся в местных технических вузах:
131, 125, 115, 122, 131, 115, 107, 99, 125, 111.
Известно, что в генеральной совокупности IQ распределен нормально.
Найдите доверительный интервал для математического ожидания с надежностью 0.95.

In [143]:
iq = np.array([131, 125, 115, 122, 131, 115, 107, 99, 125, 111])
alfa = 0.05
n = len(iq)
M = iq.mean()
sigma_n = iq.std(ddof=1)

Сигма неизвестна, находим критерий Стьюдента t от alfa/2 (т.е. t для 97.5%).

In [144]:
t = 2.262

Находим границы доверительного интервала.

In [145]:
round(M + t * (sigma_n/n**0.5), 2)

125.64

In [146]:
round(M - t * (sigma_n/n**0.5), 2)

110.56

Ответ: [110.56:125.64]

Задание 3.
Известно, что рост футболистов в сборной распределен нормально
с дисперсией генеральной совокупности, равной 25 кв.см. Объем выборки равен 27,
среднее выборочное составляет 174.2. Найдите доверительный интервал для математического
ожидания с надежностью 0.95.

In [147]:
D = 25
n = 27
M = 174.2
alfa = 0.05

Известна дисперсия, значит известна сигма. По Z таблице или с помощью библиотеки определяем Z от alfa/2 (т.е. Z для 97.5%).

In [148]:
sigma = D**0.5

In [149]:
from scipy.stats import norm

In [150]:
Z = norm.ppf(0.975)

Находим границы доверительного интервала.

In [151]:
round(M + Z * (sigma/n**0.5), 2)

176.09

In [152]:
round(M - Z * (sigma/n**0.5), 2)

172.31

Ответ: [172.31:176.09]